### PR TITLE
Corey

### PR DESCRIPTION
**Pull request by cogii**
Changed the editcourse.html file to reflect the database properly. Added in a findCourse(courseID) method to CourseManagement.py. Took the findUser() method from UserManagement.py and changed it to reflect finding a course instead of a user. Updated the editCourse(View) method in views.py. Currently running into an issues when I try to edit a course on the webpage. The error is "ValueError at /editcourse/
The view TAScheduler.views.view didn't return an HttpResponse object. It returned None instead.".  Not entirely sure what's going on.

Please ignore the views.py pull, that one was just a comment i removed when trying to test github.

### CODE REVIEW

File: TAScheduler/Management/CourseManagement.py [hunk 1]
```diff
@@ -58,6 +58,29 @@ def editCourse(Id, courseName, courseLocation, courseDays, courseHours, courseIn
...
+            if course is None:
+                raise TypeError("This ID does not exist")
+        elif not (courseID== ""):
+            try:
+                course = Course.objects.get(courseID=courseID)
+            except Course.DoesNotExist:
+                course = None
+
+            if course is None:
+                raise TypeError("This username does not exist")
```
Comment (BMHebert) [inline TAScheduler/Management/CourseManagement.py]:
```suggestion
                raise TypeError("This ID does not exist")
```
courseID should never be a string. I believe you left in some vestigial code from findUser here where it looks for a username. Also, there is no reason for course to be set to None and then later check if it is none. You can simply raise the error in the except block.

File: TAScheduler/views.py [hunk 2]
```diff
@@ -168,10 +168,24 @@ def get(request):
...
+        try:
+            edit_or_submit = request.POST["edit"]
+        except MultiValueDictKeyError:
+            edit_or_submit = request.POST["submit"]
+            edit = False
+
+        if edit:
+            change_course = CourseManagement.findCourse(courseID=edit_or_submit)
+            return render(request, "editcourse.html", {"Course_list": Course.objects.all()}, {"change_course": change_course})
+        else: CourseManagement.editCourse()
```
Comment (BMHebert) [inline TAScheduler/views.py]: The code after the else statement seems unfinished. editCourse() is not passed any arguments and a page is not rendered.

File: Templates/editcourse.html [hunk 3]
```diff
@@ -34,21 +34,34 @@ <h4>Edit Course</h4>
...
                 <th>Course Days</th>
-                <th>Course Name</th>
                 <th>Instructor Name</th>
                 <th>TA Name</th>
                 <th>Labs</th>
             </tr>
-             {% for Courses in Course %}
+             {% for Course in Course_list %}
+                 {% if change_course.courseID  == Course.courseID %}
+                     <td><input type="text" id= name=id value={{ Course.courseID }}></td>
```
Comment (BMHebert) [inline Templates/editcourse.html]:
```suggestion
                     <td><input type="number" id= name=id value={{ Course.courseID }}></td>
```

File: Templates/editcourse.html [hunk 4]
```diff
@@ -34,21 +34,34 @@ <h4>Edit Course</h4>
...
+                     <td><input type="text" id=name name=name value={{ Course.name }}></td>
+                     <td><input type="text" id=location name=location value={{ Course.location }}></td>
+                     <td><input type="text" id=hours name=hours value={{ Course.hours }}></td>
+                     <td><input type="text" id=days name=days value={{ Course.days }}></td>
+{#                     <td><input type="text" id=instructor name=instructor value={{ Course.instructor }}></td>#}
```
Comment (BMHebert) [inline Templates/editcourse.html]:
```suggestion
                     <td>
                     <select name="instructor" id="instructor">
                         {% for UserProfile in UserProfile_list &}
                             {% if UserProfile.userType == "INSTRUCTOR"
                                 <option value ={{ UserProfile.userID }}> {{ UserProfile.username }} </option>
                             {% endif %}
                         {{% endfor %}
                     </select>
                     </td>
```
Potential idea for how to implement input for instructor. This requires passing a list of all UserProfiles called UserProfile_list to be passed to the render of this page.

**Review by BMHebert — CHANGES_REQUESTED** [review]
This is mostly good, but right now it lacks the ability to actually change any data since when the editCourse() method is called, it is not passed any arguments. Also the input sections on editUser.html for instructor, TAs, and labs are currently commented out, meaning no input will be gotten for those fields.

File: Templates/editcourse.html (hunk 4 above)
Comment (cogii) [reply Templates/editcourse.html]: Added this in

File: Templates/editcourse.html (hunk 3 above)
Comment (cogii) [reply Templates/editcourse.html]: Fixed this

File: TAScheduler/views.py (hunk 2 above)
Comment (cogii) [reply TAScheduler/views.py]: Finished this part but still getting an httpresponse of none

**Review by BMHebert — APPROVED** [review]

**Pull request merged**